### ISSUE TITLE
No hover when using custom colors

### ISSUE THREAD

**Issue by paradoXp** [cakecatz/flex-toolbar#148]
When 
```cson
style:
    color: "green"
    hover: "blue"
```
is used in a button, there is no hover effect. If no color is used, the hover effect is present on the toolbar.

**Comment by UziTech** [comment]
There is [no way](https://stackoverflow.com/questions/7988097/javascript-set-an-elements-color-on-hover-through-style) to set a different style on hover with with `element.style`

If you want to submit a PR it might be helpful to add a hover property with a separate style property

**Comment by UziTech** [comment]
You could set a class with the `className` property and in your global styles.less file set a `.{className}:hover { other: styles;}`

**Comment by paradoXp** [comment]
Unfortunately I'm still trying to figure out what I would need to change so that I could open a PR. I'm still trying to learn how all the atom pieces fit together.

I'll keep trying to figure out what in the package needs to be modified, and if I do, I'll make the changes and open a PR.

**Comment by paradoXp** [comment]
One thing I noticed is that the theme **One Dark** shows hover indicators but **Atom Dark** does not. The indicator of the **One Dark** theme is a background highlight (I think). For either theme, when I set
```css
.tool-bar-btn:hover {
  color: red;
}
```
it only changes the color of the buttons that do not have
```cson
style:
    color: "{some color}"
```
I'm sorry if this is obvious, I'm not a front end guy so the behavior and structure is not intuitive for me. I'll get it eventually 😃

**Comment by UziTech** [comment]
Yes you will have use [!important](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#The_!important_exception) in your css.

```css
.tool-bar-btn:hover {
  color: red !important;
}
```

**Comment by UziTech** [comment]
If you want to add a `hover` option with different styles I could help you.

something like:

```js
{
	type: "url",
	icon: "octoface",
	url: "https://github.com/{repo-owner}/{repo-name}",
	tooltip: "Open Project Repository",
	style: {
		background: "#fff",
	},
	hover: {
		color: "#933",
		background: "#3ff",
	},
},
```

A good place to start would be to check for `btn.hover` in the `addButtons` function similar to the `btn.style` check

https://github.com/cakecatz/flex-toolbar/blob/0f71a08c3e1c9ee88d4db18537dae75b65c9ce06/lib/flex-tool-bar.js#L510-L515

Then if `btn.hover` exists add event listeners for `mouseenter` and `mouseleave` to add/remove the hover styles.

**Comment by paradoXp** [comment]
Thanks for your help! I added a PR with the change.
